Source/FieldSolver/SpectralSolver/SpectralAlgorithms/GalileanAlgorithm.cpp

```diff
@@ -174,73 +199,240 @@ void GalileanAlgorithm::InitializeSpectralCoefficients(const SpectralKSpace& spe
+                modified_ky[j]*vy + modified_kz[k]*vz;
 #else
-                                 modified_kz[j]*vz;
+                modified_kz[j]*vz;
 #endif
+            // The coefficients in the following refer to the ones given in equations
+            // (12a)-(12d) of (Lehe et al, PRE 94, 2016), used to update B and E
+            // (equations (11a) and (11b) of the same reference, respectively)
 
-                const Real nu = kv/(k_norm*c);
-                const Complex theta = amrex::exp( 0.5_rt*I*kv*dt );
-                const Complex theta_star = amrex::exp( -0.5_rt*I*kv*dt );
-                const Complex e_theta = amrex::exp( I*c*k_norm*dt );
-
-                Theta2(i,j,k) = theta*theta;
-
-                if ( (nu != 1.) && (nu != 0) ) {
-
-                    // Note: the coefficients X1, X2, X3 do not correspond
-                    // exactly to the original Galilean paper, but the
-                    // update equation have been modified accordingly so that
-                    // the expressions/ below (with the update equations)
-                    // are mathematically equivalent to those of the paper.
-                    Complex x1 = 1._rt/(1._rt-nu*nu) *
-                        (theta_star - C(i,j,k)*theta + I*kv*S_ck(i,j,k)*theta);
-                    // x1, above, is identical to the original paper
-                    X1(i,j,k) = theta*x1/(ep0*c*c*k_norm*k_norm);
-                    // The difference betwen X2 and X3 below, and those
-                    // from the original paper is the factor ep0*k_norm*k_norm
-                    X2(i,j,k) = (x1 - theta*(1._rt - C(i,j,k)))
-                                /(theta_star-theta)/(ep0*k_norm*k_norm);
-                    X3(i,j,k) = (x1 - theta_star*(1._rt - C(i,j,k)))
-                                /(theta_star-theta)/(ep0*k_norm*k_norm);
-                    X4(i,j,k) = I*kv*X1(i,j,k) - theta*theta*S_ck(i,j,k)/ep0;
-                }
-                if ( nu == 0) {
-                    X1(i,j,k) = (1._rt - C(i,j,k)) / (ep0*c*c*k_norm*k_norm);
-                    X2(i,j,k) = (1._rt - S_ck(i,j,k)/dt) / (ep0*k_norm*k_norm);
-                    X3(i,j,k) = (C(i,j,k) - S_ck(i,j,k)/dt) / (ep0*k_norm*k_norm);
-                    X4(i,j,k) = -S_ck(i,j,k)/ep0;
+            if (k_norm != 0.) {
+
+                // Auxiliary coefficients
+                const Real    k2    = k_norm * k_norm;
+                const Real    ck    = c * k_norm;
+                const Real    ckdt  = ck * dt;
+                const Complex T2aux = amrex::exp(I * ckdt); // limit of T2 for nu=1
+
+                // See equation (12a)
+                C   (i,j,k) = std::cos(ckdt);
+                S_ck(i,j,k) = std::sin(ckdt) / ck;
+
+                // See equation (12b)
+                const Real    nu      = kv / ck;
+                const Complex th      = amrex::exp(  I * 0.5_rt * kv * dt);
+                const Complex th_star = amrex::exp(- I * 0.5_rt * kv * dt);
```

Comment:
Would you mind renaming `th` and `th_star` to `theta` and `theta_star` for better readability?

Reply:
Yes, thanks for the suggestion! Done in f439e79a1eb3a20f594c244ea82a87b983688de9.